rn/Teacher/ios/Teacher/SpeedGrader/SubmissionGrader.swift

```diff
@@ -22,101 +22,158 @@ import Core
+                .padding(EdgeInsets(top: 8, leading: 16, bottom: 4, trailing: 8))
+        } else if let selected = attempts.first(where: { attempt == $0.attempt }) ?? attempts.last {
+            Button(action: { self.showAttempts.toggle() }, label: {
+                HStack {
+                    Text(selected.submittedAt?.dateTimeString ?? "")
+                        .font(.medium14)
                     Spacer()
+                    Icon.miniArrowDownSolid.rotationEffect(.degrees(showAttempts ? 180 : 0))
                 }
-                Spacer()
-                Divider()
-                VStack(spacing: 0) {
-                    Spacer().frame(height: 16)
-                    drawerContent
+                    .foregroundColor(.textDark)
+                    .padding(EdgeInsets(top: 8, leading: 16, bottom: 4, trailing: 8))
+            })
+        }
+    }
+
+    @ViewBuilder
+    var attemptPicker: some View {
+        if showAttempts {
+            VStack(spacing: 0) {
+                Picker(selection: Binding(get: { selected.attempt }, set: { newValue in
+                    withTransaction(.exclusive()) { attempt = newValue }
+                    showAttempts = false
+                }), label: Text(verbatim: "")) {
+                    ForEach(attempts.all, id: \.attempt) { attempt in
+                        Text(attempt.submittedAt?.dateTimeString ?? "")
+                            .tag(Optional(attempt.attempt))
+                    }
                 }
-                    .frame(width: 375)
+                    .labelsHidden()
+                Divider()
             }
+                .background(Color.backgroundLightest)
         }
-            .background(Color.backgroundLightest)
     }
 
-    enum DrawerTab: Int, CaseIterable, Identifiable {
+    enum GraderTab: Int, CaseIterable, Identifiable {
         case grades, comments, files
         var id: Int { rawValue }
     }
 
-    @ViewBuilder
-    var drawerContent: some View {
-        Picker(selection: Binding(get: { selectedDrawerTab }, set: {
-            selectedDrawerTab = $0
-            if drawerState == .min { drawerState = .mid }
-        }), label: Text(verbatim: "")) {
-            Text("Grades").tag(DrawerTab.grades)
-            Text("Comments").tag(DrawerTab.comments)
-            filesTabText.tag(DrawerTab.files)
-        }
-            .pickerStyle(SegmentedPickerStyle())
-            .padding(EdgeInsets(top: 0, leading: 16, bottom: 8, trailing: 16))
-        Divider()
-        Pages(items: DrawerTab.allCases, currentIndex: Binding(
-            get: { selectedDrawerTab.rawValue },
-            set: { selectedDrawerTab = DrawerTab.allCases[$0] }
-        )) { tab in
-            ScrollView {
+    var tabs: some View {
+        VStack(spacing: 0) {
+            Picker(selection: Binding(get: { tab }, set: {
+                tab = $0
+                if drawerState == .min {
+                    withTransaction(DrawerState.transaction) {
```

Comment:
needed because this is changing the parent State binding, not the drawer's copy with the transaction already on it.